clients/src/main/java/org/apache/kafka/clients/consumer/internals/NetworkClientDelegate.java

```diff
@@ -173,6 +178,7 @@ private void trySend(final long currentTimeMs) {
             unsent.timer.update(currentTimeMs);
             if (unsent.timer.isExpired()) {
                 iterator.remove();
+                kafkaConsumerMetrics.ifPresent(metrics -> metrics.recordUnsentRequestsQueueTime(this.time.milliseconds() - unsent.enqueuedMs()));
```

Comment:
shouldn't we use `currentTimeMs` instead of `this.time.milliseconds()`?

Reply:
Hi @lianetm, we may need to keep `time.milliseconds()` here.  The `currentTimeMs` is set when `runOnce` starts. However, the `enqueueTimeMs` (original `enqueuedMs`) is set when the request is added to the `unsentRequests`. It's after `currentTimeMs`. We will get negative result if using `currentTimeMs` here.